code/API_definitions/Application-endpoint-registration.yaml

```diff
@@ -0,0 +1,392 @@
+openapi: 3.0.3
+info:
+  title: Application Endpoint Registration API
+  version: 0.1.0
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  description: |
+    Application endpoint registration allows application developers to
+    register one or more Application Endpoints, and retrieve, update,and
+    delete those egistrations.
+
+    This information can we used for various usecases like optimal end
+    point discovery to help end users connect to the most most optimal
+    instance of the application. Addtionally the information can be used
```

Comment:
I feel that if can refer or qualify applications as "edge applications" then it will make it more related to edge deployments while using generically may imply any deployment including clouds. Just a thought.